pkg/server/login/login.go

```diff
@@ -166,25 +169,27 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 		failed(errorCodeAccessDenied, w, req)
 		return
 	}
-	result := metrics.SuccessResult
-	defer func() {
-		metrics.RecordFormPasswordAuth(result)
-	}()
+
 	authResponse, ok, err := l.auth.AuthenticatePassword(context.TODO(), username, password)
-	if err != nil {
+	switch {
```

Comment:
I did a switch here instead so it looks more like the basicauth.go code. I could also introduce the variant with gates instead of a switch.